source/common/http/http2/codec_impl.cc

```diff
@@ -556,7 +557,24 @@ ConnectionImpl::ConnectionImpl(Network::Connection& connection, CodecStats& stat
       protocol_constraints_(stats, http2_options),
       skip_encoding_empty_trailers_(Runtime::runtimeFeatureEnabled(
           "envoy.reloadable_features.http2_skip_encoding_empty_trailers")),
-      dispatching_(false), raised_goaway_(false), pending_deferred_reset_(false) {}
+      dispatching_(false), raised_goaway_(false), pending_deferred_reset_(false),
+      random_(random_generator) {
+  if (http2_options.has_connection_keepalive()) {
+    keepalive_interval_ = std::chrono::milliseconds(
+        PROTOBUF_GET_MS_REQUIRED(http2_options.connection_keepalive(), interval));
+    keepalive_timeout_ = std::chrono::milliseconds(
+        PROTOBUF_GET_MS_REQUIRED(http2_options.connection_keepalive(), timeout));
+    keepalive_interval_jitter_percent_ =
+        http2_options.connection_keepalive().interval_jitter_percent();
+
+    keepalive_send_timer_ = connection.dispatcher().createTimer([this]() { sendKeepalive(); });
+    keepalive_timeout_timer_ =
+        connection.dispatcher().createTimer([this]() { onKeepaliveResponseTimeout(); });
```

Comment:
Does this timer need to be cancelled in any deferred delete close paths so we don't get any kind of race? That is a typical bug that we hit.

Reply:
I grep'd through all `deferredDelete()`s in `source/common/http`, and it looks like only streams are deferDeleted, not connections.

Reply:
OK sounds good.